consensus/src/lib.rs

```diff
@@ -98,6 +98,9 @@ cfg_if::cfg_if! {
 
             /// Event indicating the container has been finalized.
             fn finalized(&mut self, proof: Proof, payload: Self::Digest) -> impl Future<Output = ()> + Send;
+
+
+            fn skipped(&mut self, proof: Proof, view: u64) -> impl Future<Output = ()> + Send;
```

Comment:
@patrick-ogrady are we aligned that this arg `view: u64` should be changed as a `payload: Self::Digest`

```
        let message = nullify_message(view);
        let digest = hash(&message);
```

Reply:
You should use `type Index` there (like `Supervisor`) 👍

Reply:
Actually...

Reply:
I'm not sure we should break anything out of `Proof`?

Seems like we could just do:
```
            fn prepared(&mut self, proof: Proof) -> impl Future<Output = ()> + Send;
            fn skipped(&mut self, proof: Proof) -> impl Future<Output = ()> + Send;
            fn finalized(&mut self, proof: Proof) -> impl Future<Output = ()> + Send;
````
 
I didn't do this before because it seemed annoying to have to parse the `Proof` (and that we should just return what the user "probably" wants...if we are assuming `Skipped` is useful then we should probably not try to optimistically break anything out.

Reply:
Huuum, yes just passing `Proof` looks enough!

Not sure it is the role of the `Prover` but it could be responsible of providing a `Digest` from a `Proof` as it has the logic of Proof parsing.
Is there any use case where prefixing the `Proof` with its type  ( an `u8` ) could be benefit ? Regarding a proof, you could identify if it is a `Finalized` `Nullified`, ...

Reply:
If we go this route, it may make sense to unify everything in `report` 🤔 ?

https://github.com/commonwarexyz/monorepo/blob/a6b5d401a6aa9d70ffbb2a51419482817540dddc/consensus/src/simplex/mocks/supervisor.rs#L89

Need to ponder a bit here.

Reply:
> If we go this route, it may make sense to unify everything in report 🤔 ?

I feel like Supervisor is essentially a synchronous component used to store/retrieve data whereas Committer is there to trigger Application logic. I suppose that the unification in `report` would give the responsability to Supervisor to send message to the Application channel, making it an orchestrator.

Reply:
Maybe it is worth to stick how it works right know? Adding the 
Committer `fn skipped(&mut self, proof: Proof, index: Self::Index) -> impl Future<Output = ()> + Send;`
and open an issue for potential refactoring on Committer/Supervisor